File: baselib/src/main/java/com/bride/baselib/BaseActivity.java
package com.bride.baselib;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;

import androidx.appcompat.app.AppCompatActivity;

/**
 * <p>Created by shixin on 2019/3/22.
 */
public class BaseActivity extends AppCompatActivity {
    private Handler mHandler = new Handler(Looper.getMainLooper());

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    protected Handler getHandler() {
        return mHandler;
    }
}

File: baselib/src/main/java/com/max/baselib/BaseActivity.kt
package com.max.baselib

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log

/**
 *
 * Created by shixin on 2018/4/26.
 */
open class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.i("create", "${this.javaClass.simpleName}; taskId=$taskId $isTaskRoot")
    }

    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        Log.i("onNewIntent", "${this.javaClass.simpleName}; taskId=$taskId $isTaskRoot")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.i("destroy", " " + this.javaClass.simpleName)
    }
}
